v-next/hardhat/src/internal/builtin-plugins/solidity-test/reporter.ts

```diff
@@ -119,33 +122,46 @@ export async function* testReporter(
     for (const [index, failure] of failures.entries()) {
       yield `\n${chalk.bold(chalk.red(`Failure (${index + 1})`))}: ${failure.name}\n`;
 
-      if (
-        failure.reason !== undefined &&
-        failure.reason !== null &&
-        failure.reason !== ""
-      ) {
-        yield `Reason: ${chalk.grey(failure.reason)}\n`;
+      const stackTrace = failure.stackTrace();
+
+      let reason: string | undefined;
+      if (stackTrace?.kind === "StackTrace") {
+        reason = getMessageFromLastStackTraceEntry(stackTrace.entries[stackTrace.entries.length - 1]);
```

Comment:
The messages from `getMessageFromLastStackTraceEntry` are written with a Hardhat Network use-case in mind and they can be a bit weird in a Solidity-testing context. There is more discussion of this here: https://github.com/NomicFoundation/hardhat/pull/6235#discussion_r1946455490

For me it's ok to merge like this, but wanted to make sure we discuss this (cc @fvictorio )

Reply:
I quickly created a duplicate of the original message inference function in solidity-test plugin and run a first pass at the error message simplification. I haven't yet fully analysed which even make sense in the context of solidiy-tests and which don't but I wanted to put it out there as quickly as possible because I think there might be value into opening up these deliberations to a wider audience. Here's the commit in question - https://github.com/NomicFoundation/hardhat/pull/6506/commits/543ea3c6862dfb1ad934325e2fb172d216e57809